glean-core/src/metrics/labeled.rs

```diff
@@ -338,8 +278,8 @@ pub fn validate_dynamic_label(
         );
         record_error(glean, meta, ErrorType::InvalidLabel, msg, None);
         true
-    } else if !matches_label_regex(label) {
-        let msg = format!("label must be snake_case, got '{}'", label);
+    } else if !label.is_ascii() || label.chars().any(|c| c.is_ascii_control()) {
```

Comment:
performance nit: this does two loops: one to check everything is ascii, one to check for control characters.
In theory this should be "faster":

```suggestion
    } else label.chars().any(|c| !c.is_ascii() || c.is_ascii_control()) {
```

but honestly for strings <= 71 chars it probably doesn't matter whatsoever.